### PR TITLE
 cleanup keychain proxy 

### PR DESCRIPTION
**Pull request by jack60612**
follow up to changes requested in 12782

### CODE REVIEW

File: chia/daemon/client.py [hunk 1]
```diff
@@ -46,25 +46,30 @@ async def start(self) -> None:
...
 
         async def listener_task() -> None:
-            await self.listener()
-            await self.close()
+            try:
+                await self.listener()
+                await self.close()
+            except Exception:
+                await self.close()
+                raise
```
Comment (altendky) [inline chia/daemon/client.py:54]:
```suggestion
            try:
                await self.listener()
            finally:
                await self.close()
```

File: chia/daemon/client.py [hunk 2]
```diff
@@ -73,23 +78,15 @@ async def _get(self, request: WsRpcMessage) -> WsRpcMessage:
...
             self.response_dict.pop(request_id)
-        else:
-            response = None  # type: ignore
-        self._request_dict.pop(request_id)
-
-        return response
+            return response
+        except asyncio.TimeoutError:
+            self._request_dict.pop(request_id)
+            raise Exception(f"No response from daemon for request_id: {request_id}.")
```
Comment (altendky) [inline chia/daemon/client.py:89]:
```suggestion
            raise Exception(f"No response from daemon for request_id: {request_id}")
```

**Review by altendky — COMMENTED** [review]
A couple quick comments, not a full review.

File: chia/daemon/client.py (hunk 2 above)
Comment (jack60612) [reply chia/daemon/client.py:89]: 👍

File: chia/daemon/client.py (hunk 1 above)
Comment (jack60612) [reply chia/daemon/client.py:54]: oh i forgot about that!

**Review by emlowe — APPROVED** [review]
Looks good

**Pull request merged**